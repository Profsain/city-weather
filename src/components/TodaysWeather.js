import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import moment from 'moment';
import { Grid } from '@mui/material';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import TextField from '@mui/material/TextField';
import fetchByCity from '../redux/fetchByCityName';
import fetchNextFiveApi from '../redux/fetchNextFive';
import setWeatherIcon from '../helper/setWeatherIcons';
import '../styles/TodaysWeather.css';

const TodaysWeather = () => {
  const weatherData = useSelector((state) => state.weather.weatherData);
  const dispatch = useDispatch();
  const weatherCondition = weatherData?.weather?.[0]?.main || 'Internet Connect Error!';
  const currentTemp = weatherData?.main?.temp || '0';
  const currentCity = weatherData?.name || 'City';
  const count = weatherData?.sys?.country || 'N';
  const date = moment().format('dddd Do MMMM YYYY');
  const weatherIcon = setWeatherIcon(weatherCondition);
  const [city, setCity] = useState('');
  const navigate = useNavigate();

  const onChangeHandler = (e) => {
    setCity(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(fetchByCity(city));
    dispatch(fetchNextFiveApi(city));
    setCity('');
  };

  const backToHome = () => {
    navigate('/');
  };

  return (
    <Grid
      className="Container"
      container
      direction="column"
      justifyContent="space-between"
      alignItems="center"
    >
      <Grid item xs={2} className="Top">
        <div className="Search-div">
          <TextField
            id="filled-basic"
            label="Seach for places"
            variant="filled"
            className="Search"
            value={city}
            onChange={onChangeHandler}
          />
          <button onClick={handleSubmit} type="button">Search</button>
        </div>
        <ArrowBackIcon
          className="Back-home"
          onClick={backToHome}
        />
      </Grid>
      <Grid item xs={6} className="Middle">
        <img src={weatherIcon} alt="wheather-today" className="WeatherIcon" />
      </Grid>
      <Grid item xs={4} className="Bottom">
        <h4 className="degree">
          {currentTemp}
          ℃
        </h4>
        <p className="Title">
          {weatherCondition || 'únknown'}
        </p>
        <p>
          Today -
          {date}
        </p>
        <p>
          {currentCity}
          -
          {count}
        </p>
      </Grid>
    </Grid>
  );
};

export default TodaysWeather;
