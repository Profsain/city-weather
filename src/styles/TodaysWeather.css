.Container {
  background-color: #1e213a;
  color: #fff;
  padding-bottom: 148px;
}

.Top {
  display: flex;
  justify-content: space-between;
  width: 100%;
  padding: 18px;
}

.Top Button {
  background-color: #6e707a;
}

.Middle {
  width: 100%;
  text-align: center;
}

.Bottom {
  text-align: center;
  width: 100%;
}

.degree {
  font-size: 48px;
}

.degree span {
  font-size: 24px;
  font-weight: 300;
}

.Title {
  font-size: 24px;
}

.WeatherIcon {
  width: 150px;
}

.Search {
  background-color: #d7d7d7;
  border-radius: 5px;
}

.Search-div {
  display: flex;
  flex-direction: column;
}

.Search-div button {
  margin-top: 4px;
  font-size: 14px;
  padding: 4px;
  border: none;
  border-radius: 4px;
  color: #fff;
  font-weight: 600;
}

.Search-div button:hover {
  cursor: pointer;
  color: #6e707a;
  background-color: #1e213a;
  border: 1px solid #fff;
}
