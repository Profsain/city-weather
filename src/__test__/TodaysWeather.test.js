import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import TodaysWeather from '../components/TodaysWeather';

it('Renders TodaysWeather component correctly', () => {
  const history = createMemoryHistory();

  const tree = renderer.create(
    <Provider store={store}>
      <Router location={history.location} navigator={history}>
        <TodaysWeather />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
