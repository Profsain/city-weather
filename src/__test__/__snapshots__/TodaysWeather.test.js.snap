// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders TodaysWeather component correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column Container css-1p4zzjm-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 Top css-1o7apob-MuiGrid-root"
  >
    <div
      className="Search-div"
    >
      <div
        className="MuiFormControl-root MuiTextField-root Search css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="filled-basic"
          id="filled-basic-label"
        >
          Seach for places
        </label>
        <div
          className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
            disabled={false}
            id="filled-basic"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium Back-home css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowBackIcon"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 Middle css-1osj8n2-MuiGrid-root"
  >
    <img
      alt="wheather-today"
      className="WeatherIcon"
      src="icons/LightCloud.png"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 Bottom css-gj1fbr-MuiGrid-root"
  >
    <h4
      className="degree"
    >
      0
      â„ƒ
    </h4>
    <p
      className="Title"
    >
      Internet Connect Error!
    </p>
    <p>
      Today -
      Saturday 6th August 2022
    </p>
    <p>
      City
      -
      N
    </p>
  </div>
</div>
`;
